linodecli/plugins/get-kubeconfig.py

```diff
@@ -0,0 +1,169 @@
+"""
+This plugin allows the option to merge an LKE Cluster's kubeconfig to a desired location.
+
+Usage:
+
+   linode-cli get-kubeconfig --id <cluster-id> --kubeconfig /path/to/config/file
+"""
+
+import argparse
+import base64
+import sys
+from pathlib import Path
+
+import yaml
+
+PLUGIN_BASE = "linode-cli get-kubeconfig"
+
+
+def call(args, context):
+    """
+    The entrypoint for this plugin
+    """
+    parser = argparse.ArgumentParser(PLUGIN_BASE, add_help=True)
+
+    group = parser.add_mutually_exclusive_group()
+
+    parser.add_argument(
+        "--kubeconfig",
+        metavar="KUBECONFIG",
+        nargs="?",
+        help="Path to kubeconfig file. If omitted, ~/.kube/config will be used.",
+        default="~/.kube/config",
+    )
+    parser.add_argument(
+        "--dry-run",
+        default=False,
+        const=True,
+        nargs="?",
+        help="Print resulting merged kubeconfig instead of merging.",
+    )
+    group.add_argument(
+        "--label",
+        metavar="LABEL",
+        nargs="?",
+        help="Label for desired cluster. If omitted, the ID must be provided.",
+    )
+    group.add_argument(
+        "--id",
+        metavar="ID",
+        nargs="?",
+        help="ID for desired cluster. If omitted, the Label must be provided.",
+    )
+
+    parsed = parser.parse_args(args)
+
+    # If --id was used, fetch the kubeconfig using the provided id
+    if parsed.id:
+        code, kubeconfig = context.client.call_operation(
+            "lke", "kubeconfig-view", args=[parsed.id]
+        )
+
+        if code != 200:
+            print(f"Error retrieving kubeconfig: {code}", file=sys.stderr)
+            sys.exit(1)
+
+    # If --label was used, fetch the kubeconfig using the provided label
+    elif parsed.label:
+        kubeconfig = _get_kubeconfig_by_label(parsed.label, context.client)
+    else:
+        print("Either --label or --id must be used.", file=sys.stderr)
+        sys.exit(1)
+
+    # Load the specified cluster's kubeconfig and the current kubeconfig
+    cluster_config = yaml.safe_load(
+        base64.b64decode(kubeconfig["kubeconfig"]).decode()
+    )
+    current_config = None
+
+    if Path(parsed.kubeconfig).expanduser().exists():
+        current_config = _load_config(Path(parsed.kubeconfig).expanduser())
+
+    # If there is no current kubeconfig, dump the cluster config to the specified file location.
+    # If there is a current kubeconfig, merge it with the cluster's kubeconfig
+    cluster_config = (
+        _merge_dict(current_config, cluster_config)
+        if current_config is not None
+        else cluster_config
+    )
+    if parsed.dry_run:
+        print(cluster_config)
+    else:
+        _dump_config(Path(parsed.kubeconfig).expanduser(), cluster_config)
```

Comment:
Can you save a path object at about line 78 and use that object instead of parsing from `str` every time?
Something like:
```python
kubeconfig_path = Path(parsed.kubeconfig).expanduser()
```

Reply:
```suggestion
    kubeconfig_path = Path(parsed.kubeconfig).expanduser()
    if Path(parsed.kubeconfig).expanduser().exists():
        current_config = _load_config(kubeconfig_path)

    # If there is no current kubeconfig, dump the cluster config to the specified file location.
    # If there is a current kubeconfig, merge it with the cluster's kubeconfig
    cluster_config = (
        _merge_dict(current_config, cluster_config)
        if current_config is not None
        else cluster_config
    )
    if parsed.dry_run:
        print(cluster_config)
    else:
        _dump_config(kubeconfig_path, cluster_config)
```